src/MainWindow.cpp

```diff
@@ -362,3 +368,29 @@ void MainWindow::disconnectSerialPortComboBox() {
 	disconnect(ui->serialPortComboBox, static_cast<void (QComboBox::*)(const QString&)>(&QComboBox::currentIndexChanged),
 	           this, &MainWindow::saveSerialPortPref);
 }
+
+void MainWindow::sendMsgButtonClicked() {
+    // if control key is not pressed, insert newline
+    if (!QGuiApplication::queryKeyboardModifiers().testFlag(Qt::ControlModifier)) {
+        ui->sendMsg_TextBox->insert("\n");
```

Comment:
According to the docs, this inserts wherever the cursor is, deleting any selected text.  Really what you want here is to also just transmit a `\n` character after sending the content of the textbox.

Reply:
Didn't realize that! Updated to append it to a QString object.